shlr/tcc/tccpp.c

```diff
@@ -1600,8 +1600,10 @@ ST_FUNC void preprocess(int is_bof)
 			memcpy (filepath, file->filename, filepath_len);
 			strcpy (filepath + filepath_len, buf);
 			if (tcc_open (s1, filepath) < 0) {
-				int len = snprintf (filepath, sizeof (filepath),
-					"/usr/include/%s", buf);
+				if (!dirname) {
+					dirname = "/usr/include";
```

Comment:
random though #1: maybe /usr/include is not where we want to look at.. maybe it's just '.' ?

Reply:
I wrote that as a fallback, in case dirname wasn't already set as the value on dir.types (it would be a null pointer in that case).
Also, "/usr/include" is the default value for dir.types. Based on C include priorities, it will prioritize a header with the same name located in the current directory than the header file located in the dir.types path

Reply:
Thats's what I thought while writing that. Might be wrong, though! :p